### PR TITLE
Crier gerrit reporter: don't log error or retry when a PR or commit i…

### PR DESCRIPTION
**Pull request by chaodaiG**
…s gone

/cc @cjwagner @mpherman2

### CODE REVIEW

File: prow/crier/reporters/gerrit/reporter.go [hunk 1]
```diff
@@ -392,7 +397,13 @@ func (c *Client) Report(ctx context.Context, logger *logrus.Entry, pj *v1.ProwJo
 			change, err = c.gc.GetChange(gerritInstance, gerritID)
 			//TODO(mpherman): In cases where the change was deleted we do not want warn nor report
```
Comment (mpherman2) [inline prow/crier/reporters/gerrit/reporter.go:398]: You can delete this now :)

File: prow/crier/reporters/gerrit/reporter.go [hunk 2]
```diff
@@ -407,31 +418,39 @@ func (c *Client) Report(ctx context.Context, logger *logrus.Entry, pj *v1.ProwJo
 	if err := c.gc.SetReview(gerritInstance, gerritID, gerritRevision, message, reviewLabels); err != nil {
 		logger.WithError(err).WithField("gerrit_id", gerritID).WithField("label", reportLabel).Info("Failed to set review.")
 
-		if reportLabel == "" {
-			return nil, nil, err
+		// It could be that the commit is deleted by the time we want to report.
+		// Swollow the error if this is the case.
+		exist, existErr := c.gc.ChangeExist(gerritInstance, gerritID)
```
Comment (mpherman2) [inline prow/crier/reporters/gerrit/reporter.go:423]: We should probably catch if existErr!=nil

**Review by cjwagner — COMMENTED** [review]
LGTM besides Mitchel's review comments.

File: prow/crier/reporters/gerrit/reporter.go (hunk 1 above)
Comment (chaodaiG) [reply prow/crier/reporters/gerrit/reporter.go]: done

File: prow/crier/reporters/gerrit/reporter.go (hunk 2 above)
Comment (chaodaiG) [reply prow/crier/reporters/gerrit/reporter.go:422]: I don' t think existErr is super useful, the outcome is the same as SetReview error. Added a log line for existErr

**Review by cjwagner — APPROVED** [review]

**Pull request merged**